src/masternodes/mn_checks.cpp

```diff
@@ -3003,6 +3010,18 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor {
         auto isPostFCE = static_cast<int>(height) >= consensus.FortCanningEpilogueHeight;
         auto isPostFCR = static_cast<int>(height) >= consensus.FortCanningRoadHeight;
         auto isPostGC  = static_cast<int>(height) >= consensus.GrandCentralHeight;
+        auto isPostNext =  static_cast<int>(height) >= consensus.NextNetworkUpgradeHeight;
+
+        if(isPostNext) {
+            const CDataStructureV0 enabledKey{AttributeTypes::Param, ParamIDs::Feature, DFIPKeys::AllowDUSDLoops};
+            auto attributes = mnview.GetAttributes();
+            assert(attributes);
+            auto DUSDLoopsAllowed= attributes->GetValue(enabledKey, false);
+            if(DUSDLoopsAllowed && hasDUSDColl && !hasOtherColl) {
```

Comment:
This should check that the loans are 100% DUSD. There is a hasDUSDLoans bool passed to CollateralPctCheck which can check if there is a DUSD loan but a hashOtherLoans or something similar would also need to be added.